packages/transaction-controller/src/TransactionController.test.ts

```diff
@@ -544,6 +571,65 @@ describe('TransactionController', () => {
       messenger = rejectMessengerMock;
     }
 
+    const mockGetNetworkClientById = jest
+      .fn()
+      .mockImplementation((networkClientId) => {
+        switch (networkClientId) {
+          // TODO(JL): This needs to use different provider from globally selected
+          case 'mainnet':
+            return {
+              configuration: {
+                chainId: toHex(1),
+              },
+              blockTracker: finalNetwork.blockTracker,
+              provider: finalNetwork.provider,
+            };
+          case 'sepolia':
+            return {
+              configuration: {
+                chainId: SEPOLIA.chainId,
+              },
+              blockTracker: buildMockBlockTracker('0x1'),
+              provider: MAINNET_PROVIDER,
+            };
+          case 'goerli':
+            return {
+              configuration: {
+                chainId: GOERLI.chainId,
+              },
+              blockTracker: buildMockBlockTracker('0x1'),
+              provider: MAINNET_PROVIDER,
+            };
+          case 'customNetworkClientId-1':
+            return {
+              configuration: {
+                chainId: '0xa',
+              },
+              blockTracker: buildMockBlockTracker('0x1'),
+              provider: MAINNET_PROVIDER,
+            };
+          case 'global':
+            return {
+              configuration: {
+                chainId: finalNetwork.state.providerConfig.chainId,
+              },
+              blockTracker: finalNetwork.blockTracker,
+              provider: finalNetwork.provider,
+            };
+          default:
+            throw new Error(`Invalid network client id ${networkClientId}`);
+        }
+      });
+
+    const mockFindNetworkClientIdByChainId = jest
+      .fn()
+      .mockImplementation((chainId) => {
+        if (chainId !== finalNetwork.state.providerConfig.chainId) {
+          throw new Error("Couldn't find networkClientId for chainId");
+        }
+        return 'global';
```

Comment:
I think we can drop returning global here and in mockGetNetworkClientById after the feature flag guards are in place

Reply:
https://github.com/MetaMask/core/pull/3643/commits/443b2f87e8b2dbb4a60319283860ecd819befd6b